### PR TITLE
[68] Be able to track user actions

### PR DESCRIPTION
**Pull request by mateuszpmroz**

### CODE REVIEW

File: src/redux/reducers/index.ts [hunk 1]
```diff
@@ -5,13 +5,16 @@ import * as currentProject from "./currentProjectReducer";
 import * as recentProjects from "./recentProjectsReducer";
 import * as appError from "./appErrorReducer";
 import * as auth from "./authReducer";
+import * as tracks from "./trackReducer";
```
Comment (bklukaczewski) [inline src/redux/reducers/index.ts]: I'm not sure about naming for `tracks` 🚚 🚚 🚚  How about `tracking`?

Comment (mateuszpmroz) [reply src/redux/reducers/index.ts]: Okay, it fits 😄

File: src/react/components/pages/editorPage/editorPage.tsx [hunk 2]
```diff
@@ -657,6 +665,16 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
             return;
         }
 
+        /**
+         * Track user leaves the image
+         */
+        if (this.state.selectedAsset && this.state.selectedAsset.asset) {
```
Comment (bklukaczewski) [inline src/react/components/pages/editorPage/editorPage.tsx]: You refer to `this.state.selectedAsset` quite often.
Maybe destructuring
```
const { selectedAsset } = this.state
```
would make it slightly more readable?

Same applies to props.

File: src/redux/reducers/trackingReducer.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        case ActionTypes.TRACK_SIGN_IN_SUCCESS:
+            return state.concat(action.payload);
+        case ActionTypes.TRACK_SIGN_OUT_SUCCESS:
+            return state.concat(action.payload);
+        case ActionTypes.TRACK_IMG_IN_SUCCESS:
+            return state.concat(action.payload);
+        case ActionTypes.TRACK_IMG_OUT_SUCCESS:
+            return state.concat(action.payload);
+        case ActionTypes.TRACK_IMG_DELETE_SUCCESS:
+            return state.concat(action.payload);
```
Comment (bklukaczewski) [inline src/redux/reducers/trackingReducer.ts]: Lifehack:
```
case ActionTypes.TRACK_SIGN_IN_SUCCESS:
case ActionTypes.TRACK_SIGN_OUT_SUCCESS:
case ActionTypes.TRACK_IMG_IN_SUCCESS:
case ActionTypes.TRACK_IMG_OUT_SUCCESS:
case ActionTypes.TRACK_IMG_DELETE_SUCCESS:
     return state.concat(action.payload);

**Pull request merged**